addons/mrp/models/stock_move.py

```diff
@@ -97,10 +97,24 @@ def _get_aggregated_product_quantities(self, **kwargs):
         """
         aggregated_move_lines = super()._get_aggregated_product_quantities(**kwargs)
         kit_name = kwargs.get('kit_name')
-        if kit_name:
-            for aggregated_move_line in aggregated_move_lines:
-                if aggregated_move_lines[aggregated_move_line]['description'] == kit_name:
+
+        to_be_removed = []
+        for aggregated_move_line in aggregated_move_lines:
+            bom = aggregated_move_lines[aggregated_move_line]['bom_id']
+            is_phantom = bom.type == 'phantom' if bom else False
+            description = aggregated_move_lines[aggregated_move_line]['description']
+
+            if kit_name:
+                if not is_phantom or bom.display_name not in kit_name:
```

Comment:
Hello, sorry for the delay !
All seems good except this condition. You compare strings so using `in` operator is not a good idea
```suggestion
                if not is_phantom or bom.display_name != kit_name:
```
Otherwise, if you have two kits, *Kit AB* and *Kit ABC*, the content of Kit AB will be included in the section of the Kit ABC